### DISCUSSION TITLE
Pole figure from Euler angle and .ctf (EBSD) file

### DISCUSSION THREAD

**Discussion by makeabhishek** [mtex-toolbox/mtex#1026] [Ask Anything] [answered]
Hi,
I have plotted the pole figure from the Euler angle obtained from EBSD data and from the .ctf file from the oxford EBSD machine.
I found that the pole figure from Euler angle is not correct, whereas the pole figure from the .ctf file matches with the pole figure obtained from the EBSD machine software.
Kindly advice.
Attached are the two pole figure and the script I used.
Pole figure using the .ctf file.
![image](https://user-images.githubusercontent.com/47937684/117839963-4d8be700-b2ae-11eb-92ac-0eefc539d475.png)
The Crystal symmetry data from the EBSD dat is
mineral        : Ti-Hex            
  color          : LightSkyBlue      
  symmetry       : 6/mmm             
  elements       : 24                
  a, b, c        : 3, 3, 4.7         
  reference frame: X||a*, Y||b, Z||c*
```
mineral_name = ebsd.CSList{1,2}.mineral; 
h=Miller({0,0,0,1},{1,0,-1,0},{2,-1,-1,0},ebsd(mineral_name).CS);
figure; plotPDF(ebsd(mineral_name).orientations,h,'contourf','smooth','figSize','medium','colorrange',[0 5]);
```

Pole figure using the Euler angle.
![image](https://user-images.githubusercontent.com/47937684/117840289-9c398100-b2ae-11eb-9bb6-4475672fe27e.png)
```
filename = 'EulerAngle.csv';
ori = orientation.load(filename,'columnNames',{'x','y','Euler1','Euler2','Euler3'},crystalSymmetry('hexagonal'));
h=[Miller(0,0,0,1,cs),Miller(1,0,-1,0,cs),Miller(2,-1,-1,0,cs)];
figure; plotPDF(ori,h,'contourf','smooth','figSize','medium','colorrange',[0 5]);
```

**Comment by zmichels** [comment]
Are the Euler angles in your ‘EulerAngle.csv’ using a Bunge convention? Perhaps this difference could be explained by using a different Euler convention than MTEX is expecting at orientation.load().

**Reply by makeabhishek** [reply]
I think it is ZX'Z'. So what should I use for that? I used Oxford nanoanalysis (AZTEC) machine and software.

**Comment by ralfHielscher** [comment]
Some questions:

1. Which options do you use to import the EBSD data?
2. Do the Euler angles in the csv and the ctf file coincide?

Ralf.

**Reply by makeabhishek** [reply]
1. Which options do you use to import the EBSD data?
    ebsd = EBSD.load(fileName);

2. Do the Euler angles in the csv and the ctf file coincide?
csv file is obtained from the same ctf file using AZTEC software (oxford instruments).
The Euler angles do not coincide.

**Reply by ralfHielscher** [reply]
If the Euler angles do not coincide isn't this rather a problem of the AZTEC software?

Ralf.

**Reply by makeabhishek** [reply]
Sorry, I misinterpret the question.
I answered in another sense that the Euler angle obtained from the .ctf file does not coincide with the Euler angle obtained from the .csv file in MTEX.

The .csv file obtained from AZTEC contains two phases (hex and cubic for Titanium64). But I can only extract orientations for the single phase in MTEX. 
How to obtain all the orientations/Euler angles from MTEX in the .csv file and compare them with the AZTEC file?

**Reply by ralfHielscher** [reply]
Your issue is still hard to understand:

1. You have Euler angles in your csv file and your ctf file. These are both text files. So you can easily compare the Euler angles in both files.
2. You can import the data into MTEX. Again you can check whether the Euler angles in the ctf file coincide with the imported Euler angles. Accordingly, you can check whether the Euler angles in the csv files coincide with the imported Euler angles. This should be the case as long you do not do any transformation and symmetry is considered correctly.
3. If the csv file mixes cubic and hex orientations MTEX should be able to know, which are cubic and which are hex. Is this information contained in the csv file?

Ralf.

**Reply by makeabhishek** [reply]
Thanks, Ralf for the detailed suggestions.

1.  I checked the file, there was a problem with the degree and radians. I changed the .csv file to a degree now and plotted the PDF. But still, it is not equally matching. 
2. Yes, the csv file mixes the cubic and hex orientations. 

PDF from the .csv file
![image](https://user-images.githubusercontent.com/47937684/117938260-6ab9c780-b339-11eb-8380-76a01ccc1320.png)
SCRIPT
```
filename = 'EulerAngle.csv';
ori = orientation.load(filename,'columnNames',{'x','y','Euler1','Euler2','Euler3'},crystalSymmetry('hexagonal'));
h=[Miller(0,0,0,1,cs),Miller(1,0,-1,0,cs),Miller(2,-1,-1,0,cs)];
figure; plotPDF(ori,h,'contourf','smooth','figSize','medium','colorrange',[0 5]);
```

PDF from the .ctf file
![image](https://user-images.githubusercontent.com/47937684/117938321-7ad1a700-b339-11eb-816b-1c86983aa475.png)
Script used for pdf plotting
```
ebsd = EBSD.load(fileName);
mineral_name = ebsd.CSList{1,2}.mineral; 
h=Miller({0,0,0,1},{1,0,-1,0},{2,-1,-1,0},ebsd(mineral_name).CS);
figure; plotPDF(ebsd(mineral_name).orientations,h,'contourf','smooth','figSize','medium','colorrange',[0 5]);
```


Regards
Abhishek

**Reply by ralfHielscher** [reply]
You forgot to answer to question 3. Also, you did not answer whether the Euler angles are the same after you changed from radiant to degree. 
```ebsd.rotations(1:10)```
are the first 10 Euler angles of the EBSD data. Are those the same as 
```ori(1:10)```
Do they coincide with the Euler angles in the files?

Ralf.

**Reply by makeabhishek** [reply]
Hi Ralf, I answer the questions sequentially now.
**1. You have Euler angles in your csv file and your ctf file. These are both text files. So you can easily compare the Euler angles in both files.**
A screenshot of both csv and ctf file is here.
![image](https://user-images.githubusercontent.com/47937684/118102472-6d84ed00-b40b-11eb-9ec7-37380dab0895.png)

**2. You can import the data into MTEX. Again you can check whether the Euler angles in the ctf file coincide with the imported Euler angles. Accordingly, you can check whether the Euler angles in the csv files coincide with the imported Euler angles. This should be the case as long you do not do any transformation and symmetry is considered correctly.**
This is discussed in previous message.

**3. If the csv file mixes cubic and hex orientations MTEX should be able to know, which are cubic and which are hex. Is this information contained in the csv file?**
Yes, the CSV file mixes cubic and hex orientations. I have attached the screenshot of CSV file in first message which has (x, y, Euler1,Euler2, Euler3)


The data is not the same.
```
ebsd.rotations(1:10)
 ans = rotation (show methods, plot)
  size: 10 x 1
 
  Bunge Euler angles in degree
  phi1  Phi phi2 Inv.
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
   360  180    0    0
```
from CSV data
```
CSV_data(1:10,:)
ans =
     0     0     0     0     0
     1     0     0     0     0
     2     0     0     0     0
     3     0     0     0     0
     4     0     0     0     0
     5     0     0     0     0
     6     0     0     0     0
     7     0     0     0     0
     8     0     0     0     0
     9     0     0     0     0
```

What is the difference between orientations and rotations?

**Reply by makeabhishek** [reply]
Hi Ralf,
May I please get your suggestion on the above query?

Regards
Abhishek

**Reply by ralfHielscher** [reply]
I think the fastest way would be if you send me the two files.

Ralf.

**Comment by makeabhishek** [comment]
Thanks Ralf,
May I please get your email. I am not allowed to share that data online.

Best Regards
Abhishek

**Reply by ralfHielscher** [reply]
It is seconds of internet search to find my email ....

**Comment by ralfHielscher** [comment, marked as answer]
The point is that your orientation data contain l lot of zero lines. I guess those need to be thrown away.
The following code result in almost identical pole figures on my computer.

```matlab
% import the data 
ebsd = EBSD.load('TiSample22P.ctf')
csHex = ebsd('Ti-Hex').CS;
ori = orientation.load('Euler22P.csv','ColumnNames',{'x','y','Euler1','Euler2','Euler3'},'radians',csHex)

% remove zero lines
ori = ori(ori ~= orientation.id(cs));
```
Ralf.
